### DISCUSSION TITLE
Wrong voltage displayed after calibration / not possible to calibrate [Gosund SP112]

### DISCUSSION THREAD

**Discussion by mircsicz** [arendst/Tasmota#19105] [Q&A] [answered]
### PROBLEM DESCRIPTION
I've got a set of four Gosund SP112 from roughly 2020/2021, had them in the shelf for a while and just wanted to put em to use now. After following the procedure from the [Calibration-Procedure](https://tasmota.github.io/docs/Power-Monitoring-Calibration/#calibration-procedure) doc I don't get correct values.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): Gosund SP112 (probably v3.4 as it's dated 02/2021)
- [x] Tasmota binary firmware version number used: 13.0.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua

10:49:41.067 CMD: Backlog Template; Module; GPIO 255
10:49:41.088 RSL: RESULT = {"NAME":"AMR-SERVER","GPIO":[0,576,0,32,2720,2656,0,0,2624,320,224,0,0,0],"FLAG":0,"BASE":55}
10:49:41.315 RSL: RESULT = {"Module":{"0":"AMR-SERVER"}}
10:49:41.521 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"576":"LedLink_i"},"GPIO2":{"0":"None"},"GPIO3":{"32":"Button1"},"GPIO4":{"2720":"BL0937 CF"},"GPIO5":{"2656":"HLWBL CF1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"2624":"HLWBL SEL_i"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"224":"Relay1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
10:50:23.286 CMD: Status 0
10:50:23.293 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"AMR-SP112-03","FriendlyName":["AMR-SP112-03-SERVER"],"Topic":"tasmota_900F55","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":0,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
10:50:23.326 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:06:42","StartupUTC":"2023-07-13T09:43:41","Sleep":50,"CfgHolder":4617,"BootCount":10,"BCResetTime":"2023-07-13T10:27:43","SaveCount":29,"SaveAddress":"F6000"}}
10:50:23.359 RSL: STATUS2 = {"StatusFWR":{"Version":"13.0.0(tasmota)","BuildDateTime":"2023-06-26T10:42:07","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"366/699"}}
10:50:23.378 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["J&M IoT",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A002800000000","00000080","00006000","00004000","00000000"]}}
10:50:23.413 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":634,"Free":368,"Heap":24,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,
![Bildschirmfoto 2023-07-13 um 12 00 22](https://github.com/arendst/Tasmota/assets/7398305/71a6a033-9eee-4a15-9ec7-cb409bb96cff)
20,21,22,24,26,27,29,30,35,37,45,62,68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
10:50:23.446 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-900F55-3925","IPAddress":"10.10.190.73","Gateway":"10.10.190.1","Subnetmask":"255.255.255.0","DNSServer1":"10.10.190.1","DNSServer2":"0.0.0.0","Mac":"70:03:9F:90:0F:55","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
10:50:23.469 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_900F55","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
10:50:23.491 RSL: STATUS7 = {"StatusTIM":{"UTC":"2023-07-13T09:50:23","Local":"2023-07-13T10:50:23","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"05:00","Sunset":"20:50"}}
10:50:23.507 RSL: STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
10:50:23.526 RSL: STATUS10 = {"StatusSNS":{"Time":"2023-07-13T10:50:23","ENERGY":{"TotalStartTime":"2023-07-13T10:27:43","Total":0.034,"Yesterday":0.002,"Today":0.033,"Power":53,"ApparentPower":152,"ReactivePower":143,"Factor":0.35,"Voltage":368,"Current":0.413}}}
10:50:23.546 RSL: STATUS11 = {"StatusSTS":{"Time":"2023-07-13T10:50:23","Uptime":"0T00:06:42","UptimeSec":402,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"J&M IoT","BSSId":"7A:45:58:A2:E2:7D","Channel":1,"Mode":"11n","RSSI":40,"Signal":-80,"LinkCount":1,"Downtime":"0T00:00:03"}}}
```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE

Plug device into my german LIDL Kill-a-Watt and plug a 60W Light-buld into the SP122 open console and main menu then I read and put the following commands:
`
savedata 1
PowerSet 59.0
VoltageSet 220
CurrentSet 263.636
savedata 0
`
after that procedure I plug the dev to it's final location, in my case Washer, Drayer and server. When reconnecting they show weird readings like in the attachment

### EXPECTED BEHAVIOUR
Showing approx 220 Volts instead of 350ish with current & Power according to that...

### SCREENSHOTS
![Bildschirmfoto 2023-07-13 um 11 54 42](https://github.com/arendst/Tasmota/assets/7398305/5936c808-9305-4d96-bac0-bfcec21ab95d)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

I've started the process with that [article](https://templates.blakadder.com/gosund_SP112_v3_4.html) and used the template mentioned there and also the "Gosund SP1 v23 (55)" template which is part of the distribution. I've now done the whole procedure with three of my 4 devices... I've one left that still is on 9.2.0 from 2021

I've now spend the better part of the morning reading through anything I could find about the issue and calibrating the dev. Even the old stuff: https://github.com/arendst/Tasmota/issues/5756

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by arendst** [comment]
Where does the ``savedata 0`` come from? It defeats your calibration!

**Comment by mircsicz** [comment]
@arendst THX for the awesome fast response! That came from here: https://www.tasmota.info/sp112-flash/#messwerte 

And just to be clear I've started with the doc's so without those savedata params...

**Reply by Jason2866** [reply, marked as answer]
Don't know what you are doing different as described in the documentation. 
Calibrated yesterday a new power monitoring device like it is described. Just works.
Do a `reset 5` or `reset 6` and start from scratch. Don't do this `savedata` stuff.
Tasmota has decent default settings to take care of flash. Really no need to do changes!

**Reply by Jason2866** [reply]
There a lot of tutorial and videos outdated or from people not really knowing what the are doing.
There is just one actual documentation. The official one. `Tasmota.com`
There is no other site, which is officially affiliated to Tasmota.

**Reply by mircsicz** [reply]
Wanted to ask for reset params but found em:

https://tasmota.github.io/docs/Commands/#decode-config-tool

1 = reset device settings to firmware defaults and restart (see warning below)
2 = erase flash, reset device settings to firmware defaults and restart
3 = erase System Parameter Area in flash (Wi-Fi calibration and related data) and restart (see warning below)
4 = reset device settings to firmware defaults but retain Wi-Fi credentials and restart
5 = erase all flash and reset parameters to firmware defaults but keep Wi-Fi settings and restart
6 = erase all flash and reset parameters to firmware defaults but keep Wi-Fi and MQTT settings and restart
(Erase of flash can take a few seconds to complete and there is no output during the erase process on the serial or web console)
99 = reset device bootcount to zero

Will do a "reset 6" and report back

**Comment by mircsicz** [comment]
Just to be sure if it really is a v3.4 I've opened it and took a picture
![IMG_6732](https://github.com/arendst/Tasmota/assets/7398305/fd8ad28d-7aaa-41a3-bf8a-9d2c813a91c7)

**Comment by mircsicz** [comment]
After starting over with the template from:  https://templates.blakadder.com/gosund_SP112_v3_4.html using the template for the new version:

`
{"NAME":"Gosund 112v3.4","GPIO":[56,0,57,0,132,134,0,0,131,30,21,0,0],"FLAG":4,"BASE":18}
`

and recalibrating all three devices it works now as expected.
